File: app/src/main/java/com/chris/eban/data/EventDao.java
package com.chris.eban.data;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public abstract class EventDao {

    /*************** 查询 ***************/
    @Query("select * from " + Tables.EVENT.TABLE_NAME + " order by id desc")
    public abstract List<Event> queryEventList();


    /****************  插入 ***************/
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public abstract long insertEvent(Event event);

    /****************  更新  **************/
    @Update(onConflict = OnConflictStrategy.REPLACE)
    public abstract void updateEvent(Event event);

    /****************  删除  **************/
    @Delete
    public abstract void deleteEvent(Event event);

}

File: app/src/main/java/com/chris/eban/data/EventDao.kt
package com.chris.eban.data

import androidx.room.*

@Dao
abstract class EventDao {

    /***************  查询   ***************************/

    @Query("SELECT * FROM " + Tables.EVENT.TABLE_NAME + " ORDER BY id DESC")
    abstract fun queryEventList(): List<DataEvent>

    @Query("SELECT * FROM " + Tables.EVENT.TABLE_NAME + " WHERE id = :itemId ")
    abstract fun queryEventItem(itemId: Long): DataEvent

    /****************  插入  **************************/
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    abstract fun insertEvent(event: DataEvent): Long

    /****************  更新   *************************/
    @Update(onConflict = OnConflictStrategy.REPLACE)
    abstract fun updateEvent(event: DataEvent)

    /****************  删除   *************************/
    @Delete
    abstract fun deleteEvent(event: DataEvent)

    abstract fun insertEvent(event: Event): Long

    /****************  更新   *************************/
    @Update(onConflict = OnConflictStrategy.REPLACE)
    abstract fun updateEvent(event: Event)

    /****************  删除   *************************/
    @Delete
    abstract fun deleteEvent(event: Event)

}
